### PR TITLE
Samples: Some reorg and addition of a new sample

### PR DESCRIPTION
**Pull request by nashif**
New sample to demonstrate the use of KBUILD_ZEPHYR_APP

### CODE REVIEW

File: samples/mpu/index.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.. _mpu--samples:
```
Comment (dbkinder) [inline samples/mpu/index.rst]: mpu-samples

File: samples/mpu/index.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+MMU/MPU Samples
+###############
+
+.. toctree::
+   :maxdepth: 1
+   :glob:
+
+   */*
```
Comment (dbkinder) [inline samples/mpu/index.rst]: might use **/* as you did in the others

**Review by dbkinder — CHANGES_REQUESTED** [review]
One doc title and toctree fix. Hard to see what the final doc organization would look like without a local doc build, but looks OK.  I'll verify after the merge.

File: samples/mpu/index.rst (hunk 1 above)
Comment (nashif) [inline samples/mpu/index.rst]: why mmu? where is this coming from?

Comment (nashif) [inline samples/mpu/index.rst]: ah, like the title...

Comment (dbkinder) [inline samples/mpu/index.rst]: :)

**Pull request merged**